### DISCUSSION TITLE
Vercel的Serverless Functions Execution 用量超限

### DISCUSSION THREAD

**Discussion by pseudoxx** [spencerwooo/onedrive-vercel-index#633] [Q&A] [answered]
如题 Vercel免费计划中此项的限制是100GB-Hrs每月。
自己搭了一个下载中转站，里面的文件都是200M-1G的不可预览文件。
日访问量大约在1000人次，人均使用量大，redis限制20个connections常常爆满。
只过了七天就超限了，已经被ban了，笑死。
各路大神有没有什么高招压一下带宽用量？
用量如下图
![48AD819B-DF19-4BEB-A147-815B4A5F383F](https://user-images.githubusercontent.com/41193900/166729739-fe1e4d31-2a9f-468a-ba3b-d718a86adfba.jpeg)
目前的思路：套了一层Cloudflare全站缓存，正在做对照实验，明天才能看到结果。

**Comment by zkeq** [comment, marked as answer]
目前我使用的是 `Replit` + `Vercel` 反代

朋友写的 `replit` 部署教程仓库（本地的redis）：https://github.com/valetzx/onedrivereplindex

反代 `vercel.json` 配置：
```json
{
    "version": 2,
    "routes": [
        {
            "src": "/(.*)",
            "dest": "https://域名/$1",
            "headers": {
                "cache-control": "ms-maxage=1000"
            }
        }
    ]
  }
```

我认为这是个很理想的解决方案

```json
https://tuostudy.com
```

![333](https://user-images.githubusercontent.com/62864752/166414167-c54272bc-a56c-486f-b33f-c6323afaf899.png)

**Reply by zkeq** [reply]
`vercel` 的用量很少，就反代的一点流量..

而且不用担心 `redis` 的费用问题了

并且解决了 `搜索总是突然白屏的问题`

（vercel 的 api 在短时间内调用次数过多）

**Reply by pseudoxx** [reply]
我研究一下，感谢指点！

**Reply by pseudoxx** [reply]
我发现一些问题：
1.我这边的Replit似乎没办法完成部署（一直卡在 pnpm build的一步 RAM和CPU爆满）
2.Replit似乎不能保持Always Online的状态…

**Reply by zkeq** [reply]
> 我发现一些问题： 1.我这边的Replit似乎没办法完成部署（一直卡在 pnpm build的一步 RAM和CPU爆满） 2.Replit似乎不能保持Always Online的状态…

1. 建议多等一会，`ovi` 的全部依赖有 800MB 左右
2. 建议使用监控程序保活（定时访问）

保活示例
```
https://up.repl.link/
https://fp.jacebapps.com/
```

**Reply by pseudoxx** [reply]
👌已全部解决 非常感谢！

**Reply by pseudoxx** [reply]
新问题出现了
![5CA07EBB-87B5-49F0-AED5-81EB77B369E2](https://user-images.githubusercontent.com/41193900/167084207-6a387cca-a665-483e-a73f-11d9e444dcbe.jpeg)
一段时间后Console会显示这个 然后网页502错误…必须重启replit…
看了一下这个路径的文件似乎是一个Vercel的logo图片 不知道为什么…

**Reply by pseudoxx** [reply]
> 

大概意思是无法找到文件…

**Reply by zkeq** [reply]
我也遇到了，而且遇到了很多次，最后我手动创建了一个文件。
叫那个名字，一直到现在还没崩（

（那个文件夹好像是个空文件夹）

**Reply by pseudoxx** [reply]
我去试试！感谢指点（

**Reply by pseudoxx** [reply]
新问题又出现了：过几个小时我创建的占位文件会被自动删掉，然后又报502错误……

**Reply by pseudoxx** [reply]
这也太神奇了！

**Reply by pseudoxx** [reply]
现在暂时在用freshping监测502错误 希望过一段时间之后可以稳定下来…

**Reply by zkeq** [reply]
> 新问题又出现了：过几个小时我创建的占位文件会被自动删掉，然后又报502错误……

这个的解决办法是

在修改文件之后，重启一下项目（`SOTP` 后 `RUN`）

`replit` 确实存在回档的情况

**Reply by pseudoxx** [reply]
学到了！

**Reply by weekdaycare** [reply]
我这边部署完成之后可以预览，但是进网页就显示`The repl is having some trouble waking up. Please try again in 30 seconds.`请问是哪里出了问题呀？
网站地址[https://oneindex.weekdaycare.repl.co](https://oneindex.weekdaycare.repl.co)
![image](https://user-images.githubusercontent.com/63026074/175781296-e3ef4636-7f95-4f68-9608-d735405ed308.png)

**Comment by xlenco** [comment]
@orgsss 前一阵子刚研究明白Replit部署onedrive-vercel-index，[dome](https://xilej.eu.org)
2.Replit似乎不能保持Always Online的状... ,这个有办法
加我q 1043865083可详细说

**Comment by xlenco** [comment]
@orgsss
